src/handlers/web3.ts

```diff
@@ -833,3 +836,25 @@ export const estimateGasLimit = async (
     return estimateGas(estimateGasData, provider);
   }
 };
+
+const hasMergedStorage = new MMKV();
+export const getHasMerged = () =>
+  hasMergedStorage.getBoolean(STORAGE_IDS.HAS_MERGED);
+export const setHasMerged = () =>
+  hasMergedStorage.set(STORAGE_IDS.HAS_MERGED, true);
+export const setHasNotMerged = () =>
+  hasMergedStorage.set(STORAGE_IDS.HAS_MERGED, false);
+export const checkForTheMerge = async (provider: JsonRpcProvider) => {
+  const block = await provider.getBlock('latest');
+  logger.debug('LATEST BLOCK: ', block);
+  const { _difficulty } = block;
+  logger.debug('_DIFFICULTY: ', _difficulty.toString());
+  const currentHasMerged = getHasMerged();
+  logger.debug('CURRENT HAS MERGED: ', currentHasMerged);
+  if (_difficulty.toString() === '0') {
+    setHasMerged();
+  } else if (currentHasMerged) {
+    setHasNotMerged();
+  }
+  logger.debug('GET HAS MERGED FINAL: ', getHasMerged());
+};
```

Comment:
This works nicely when toggling between mainnet and ropsten.

There are 2 outstanding questions with regard to the flag:
1. Does it make sense to make a hook for this? So far I cannot think of a reason why the view layer will need to access this flag.
2. Should we be polling here instead?